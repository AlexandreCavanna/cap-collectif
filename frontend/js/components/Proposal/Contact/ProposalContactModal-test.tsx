/* eslint-env jest */
import * as React from 'react'
import { shallow } from 'enzyme'
import { ProposalContactModal } from './ProposalContactModal'

describe('<ProposalContactModal />', () => {
  it('renders correctly', () => {
    const props = {
      proposalId: 'proposalId324Z',
      authorName: 'Gertrude',
      onClose: jest.fn(),
      addCaptchaField: false,
      show: false,
    }
    const wrapper = shallow(<ProposalContactModal {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
