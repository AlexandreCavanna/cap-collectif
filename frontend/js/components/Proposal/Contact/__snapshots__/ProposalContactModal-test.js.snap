// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProposalContactModal /> renders correctly 1`] = `
<Modal
  animation={true}
  aria-labelledby="ProposalFormContactModal-modal"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <form
    id="ProposalContactModalForm"
    onSubmit={[Function]}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="close.modal"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
        id="ProposalFormContactModal-title"
      >
        <FormattedMessage
          id="send-message-to"
          values={
            Object {
              "messageTo": "Gertrude",
            }
          }
        />
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <FormControl
        control={
          Object {
            "_defaultValues": Object {
              "proposalId": "proposalId324Z",
            },
            "_executeSchema": [Function],
            "_fields": Object {},
            "_focusError": [Function],
            "_formState": Object {
              "dirtyFields": Object {},
              "errors": Object {},
              "isDirty": false,
              "isLoading": true,
              "isSubmitSuccessful": false,
              "isSubmitted": false,
              "isSubmitting": false,
              "isValid": false,
              "isValidating": false,
              "submitCount": 0,
              "touchedFields": Object {},
            },
            "_formValues": Object {
              "proposalId": "proposalId324Z",
            },
            "_getDirty": [Function],
            "_getFieldArray": [Function],
            "_getWatch": [Function],
            "_names": Object {
              "array": Array [],
              "mount": Array [],
              "unMount": Array [],
              "watch": Array [],
            },
            "_options": Object {
              "defaultValues": Object {
                "proposalId": "proposalId324Z",
              },
              "mode": "onChange",
              "reValidateMode": "onChange",
              "shouldFocusError": true,
            },
            "_proxyFormState": Object {
              "dirtyFields": false,
              "errors": false,
              "isDirty": false,
              "isSubmitting": "all",
              "isValid": "all",
              "isValidating": false,
              "touchedFields": false,
            },
            "_removeUnmounted": [Function],
            "_reset": [Function],
            "_stateFlags": Object {
              "action": false,
              "mount": false,
              "watch": false,
            },
            "_subjects": Object {
              "array": Object {
                "next": [Function],
                "observers": Array [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "state": Object {
                "next": [Function],
                "observers": Array [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "watch": Object {
                "next": [Function],
                "observers": Array [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
            },
            "_updateFieldArray": [Function],
            "_updateFormState": [Function],
            "_updateValid": [Function],
            "getFieldState": [Function],
            "register": [Function],
            "unregister": [Function],
          }
        }
        isRequired={true}
        name="senderName"
      >
        <FormLabel
          htmlFor="ProposalFormContactModal-senderName"
          label="your-name"
        />
        <ForwardRef
          control={
            Object {
              "_defaultValues": Object {
                "proposalId": "proposalId324Z",
              },
              "_executeSchema": [Function],
              "_fields": Object {},
              "_focusError": [Function],
              "_formState": Object {
                "dirtyFields": Object {},
                "errors": Object {},
                "isDirty": false,
                "isLoading": true,
                "isSubmitSuccessful": false,
                "isSubmitted": false,
                "isSubmitting": false,
                "isValid": false,
                "isValidating": false,
                "submitCount": 0,
                "touchedFields": Object {},
              },
              "_formValues": Object {
                "proposalId": "proposalId324Z",
              },
              "_getDirty": [Function],
              "_getFieldArray": [Function],
              "_getWatch": [Function],
              "_names": Object {
                "array": Array [],
                "mount": Array [],
                "unMount": Array [],
                "watch": Array [],
              },
              "_options": Object {
                "defaultValues": Object {
                  "proposalId": "proposalId324Z",
                },
                "mode": "onChange",
                "reValidateMode": "onChange",
                "shouldFocusError": true,
              },
              "_proxyFormState": Object {
                "dirtyFields": false,
                "errors": false,
                "isDirty": false,
                "isSubmitting": "all",
                "isValid": "all",
                "isValidating": false,
                "touchedFields": false,
              },
              "_removeUnmounted": [Function],
              "_reset": [Function],
              "_stateFlags": Object {
                "action": false,
                "mount": false,
                "watch": false,
              },
              "_subjects": Object {
                "array": Object {
                  "next": [Function],
                  "observers": Array [],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                },
                "state": Object {
                  "next": [Function],
                  "observers": Array [],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                },
                "watch": Object {
                  "next": [Function],
                  "observers": Array [],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                },
              },
              "_updateFieldArray": [Function],
              "_updateFormState": [Function],
              "_updateValid": [Function],
              "getFieldState": [Function],
              "register": [Function],
              "unregister": [Function],
            }
          }
          id="ProposalFormContactModal-senderName"
          minLength={2}
          name="senderName"
          type="text"
        />
      </FormControl>
      <FormControl
        control={
          Object {
            "_defaultValues": Object {
              "proposalId": "proposalId324Z",
            },
            "_executeSchema": [Function],
            "_fields": Object {},
            "_focusError": [Function],
            "_formState": Object {
              "dirtyFields": Object {},
              "errors": Object {},
              "isDirty": false,
              "isLoading": true,
              "isSubmitSuccessful": false,
              "isSubmitted": false,
              "isSubmitting": false,
              "isValid": false,
              "isValidating": false,
              "submitCount": 0,
              "touchedFields": Object {},
            },
            "_formValues": Object {
              "proposalId": "proposalId324Z",
            },
            "_getDirty": [Function],
            "_getFieldArray": [Function],
            "_getWatch": [Function],
            "_names": Object {
              "array": Array [],
              "mount": Array [],
              "unMount": Array [],
              "watch": Array [],
            },
            "_options": Object {
              "defaultValues": Object {
                "proposalId": "proposalId324Z",
              },
              "mode": "onChange",
              "reValidateMode": "onChange",
              "shouldFocusError": true,
            },
            "_proxyFormState": Object {
              "dirtyFields": false,
              "errors": false,
              "isDirty": false,
              "isSubmitting": "all",
              "isValid": "all",
              "isValidating": false,
              "touchedFields": false,
            },
            "_removeUnmounted": [Function],
            "_reset": [Function],
            "_stateFlags": Object {
              "action": false,
              "mount": false,
              "watch": false,
            },
            "_subjects": Object {
              "array": Object {
                "next": [Function],
                "observers": Array [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "state": Object {
                "next": [Function],
                "observers": Array [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "watch": Object {
                "next": [Function],
                "observers": Array [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
            },
            "_updateFieldArray": [Function],
            "_updateFormState": [Function],
            "_updateValid": [Function],
            "getFieldState": [Function],
            "register": [Function],
            "unregister": [Function],
          }
        }
        isRequired={true}
        name="replyEmail"
      >
        <FormLabel
          htmlFor="ProposalFormContactModal-replyEmail"
          label="your-email-address"
        />
        <ForwardRef
          control={
            Object {
              "_defaultValues": Object {
                "proposalId": "proposalId324Z",
              },
              "_executeSchema": [Function],
              "_fields": Object {},
              "_focusError": [Function],
              "_formState": Object {
                "dirtyFields": Object {},
                "errors": Object {},
                "isDirty": false,
                "isLoading": true,
                "isSubmitSuccessful": false,
                "isSubmitted": false,
                "isSubmitting": false,
                "isValid": false,
                "isValidating": false,
                "submitCount": 0,
                "touchedFields": Object {},
              },
              "_formValues": Object {
                "proposalId": "proposalId324Z",
              },
              "_getDirty": [Function],
              "_getFieldArray": [Function],
              "_getWatch": [Function],
              "_names": Object {
                "array": Array [],
                "mount": Array [],
                "unMount": Array [],
                "watch": Array [],
              },
              "_options": Object {
                "defaultValues": Object {
                  "proposalId": "proposalId324Z",
                },
                "mode": "onChange",
                "reValidateMode": "onChange",
                "shouldFocusError": true,
              },
              "_proxyFormState": Object {
                "dirtyFields": false,
                "errors": false,
                "isDirty": false,
                "isSubmitting": "all",
                "isValid": "all",
                "isValidating": false,
                "touchedFields": false,
              },
              "_removeUnmounted": [Function],
              "_reset": [Function],
              "_stateFlags": Object {
                "action": false,
                "mount": false,
                "watch": false,
              },
              "_subjects": Object {
                "array": Object {
                  "next": [Function],
                  "observers": Array [],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                },
                "state": Object {
                  "next": [Function],
                  "observers": Array [],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                },
                "watch": Object {
                  "next": [Function],
                  "observers": Array [],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                },
              },
              "_updateFieldArray": [Function],
              "_updateFormState": [Function],
              "_updateValid": [Function],
              "getFieldState": [Function],
              "register": [Function],
              "unregister": [Function],
            }
          }
          id="ProposalFormContactModal-replyEmail"
          name="replyEmail"
          type="email"
        />
      </FormControl>
      <FormControl
        control={
          Object {
            "_defaultValues": Object {
              "proposalId": "proposalId324Z",
            },
            "_executeSchema": [Function],
            "_fields": Object {},
            "_focusError": [Function],
            "_formState": Object {
              "dirtyFields": Object {},
              "errors": Object {},
              "isDirty": false,
              "isLoading": true,
              "isSubmitSuccessful": false,
              "isSubmitted": false,
              "isSubmitting": false,
              "isValid": false,
              "isValidating": false,
              "submitCount": 0,
              "touchedFields": Object {},
            },
            "_formValues": Object {
              "proposalId": "proposalId324Z",
            },
            "_getDirty": [Function],
            "_getFieldArray": [Function],
            "_getWatch": [Function],
            "_names": Object {
              "array": Array [],
              "mount": Array [],
              "unMount": Array [],
              "watch": Array [],
            },
            "_options": Object {
              "defaultValues": Object {
                "proposalId": "proposalId324Z",
              },
              "mode": "onChange",
              "reValidateMode": "onChange",
              "shouldFocusError": true,
            },
            "_proxyFormState": Object {
              "dirtyFields": false,
              "errors": false,
              "isDirty": false,
              "isSubmitting": "all",
              "isValid": "all",
              "isValidating": false,
              "touchedFields": false,
            },
            "_removeUnmounted": [Function],
            "_reset": [Function],
            "_stateFlags": Object {
              "action": false,
              "mount": false,
              "watch": false,
            },
            "_subjects": Object {
              "array": Object {
                "next": [Function],
                "observers": Array [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "state": Object {
                "next": [Function],
                "observers": Array [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "watch": Object {
                "next": [Function],
                "observers": Array [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
            },
            "_updateFieldArray": [Function],
            "_updateFormState": [Function],
            "_updateValid": [Function],
            "getFieldState": [Function],
            "register": [Function],
            "unregister": [Function],
          }
        }
        isRequired={true}
        name="message"
      >
        <FormLabel
          htmlFor="ProposalFormContactModal-message"
          label="contact.your-message"
        />
        <ForwardRef
          control={
            Object {
              "_defaultValues": Object {
                "proposalId": "proposalId324Z",
              },
              "_executeSchema": [Function],
              "_fields": Object {},
              "_focusError": [Function],
              "_formState": Object {
                "dirtyFields": Object {},
                "errors": Object {},
                "isDirty": false,
                "isLoading": true,
                "isSubmitSuccessful": false,
                "isSubmitted": false,
                "isSubmitting": false,
                "isValid": false,
                "isValidating": false,
                "submitCount": 0,
                "touchedFields": Object {},
              },
              "_formValues": Object {
                "proposalId": "proposalId324Z",
              },
              "_getDirty": [Function],
              "_getFieldArray": [Function],
              "_getWatch": [Function],
              "_names": Object {
                "array": Array [],
                "mount": Array [],
                "unMount": Array [],
                "watch": Array [],
              },
              "_options": Object {
                "defaultValues": Object {
                  "proposalId": "proposalId324Z",
                },
                "mode": "onChange",
                "reValidateMode": "onChange",
                "shouldFocusError": true,
              },
              "_proxyFormState": Object {
                "dirtyFields": false,
                "errors": false,
                "isDirty": false,
                "isSubmitting": "all",
                "isValid": "all",
                "isValidating": false,
                "touchedFields": false,
              },
              "_removeUnmounted": [Function],
              "_reset": [Function],
              "_stateFlags": Object {
                "action": false,
                "mount": false,
                "watch": false,
              },
              "_subjects": Object {
                "array": Object {
                  "next": [Function],
                  "observers": Array [],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                },
                "state": Object {
                  "next": [Function],
                  "observers": Array [],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                },
                "watch": Object {
                  "next": [Function],
                  "observers": Array [],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                },
              },
              "_updateFieldArray": [Function],
              "_updateFormState": [Function],
              "_updateValid": [Function],
              "getFieldState": [Function],
              "register": [Function],
              "unregister": [Function],
            }
          }
          id="ProposalFormContactModal-message"
          name="message"
          type="textarea"
        />
      </FormControl>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <Button
        mr={2}
        onClick={[Function]}
        variant="secondary"
        variantColor="hierarchy"
        variantSize="medium"
      >
        global.cancel
      </Button>
      <Button
        disabled={true}
        id="ProposalFormContactModal-submit"
        isLoading={false}
        type="submit"
        variant="primary"
        variantSize="medium"
      >
        global.validate
      </Button>
    </ModalFooter>
  </form>
</Modal>
`;
